### PR TITLE
PWRS5PZ-24: Friend testing refactor

### PR DESCRIPTION
**Pull request by SpectraL519**
* Replaced friend testing function declarations defined in the ap/argument_parser.hpp header file with test fixture classes friend declarations
* Modify the unit tests to use created text fixtures
* Delete the `test/testing_friend_definitions.hpp` file
* Moved all test cases to a namespace ap_testing (test fixtures should be defined in this namespace)

### CODE REVIEW

File: test/include/argument_parser_test_fixture.hpp [hunk 1]
```diff
@@ -80,7 +80,8 @@ struct argument_parser_test_fixture {
         );
     }
 
-    void add_arguments(ap::argument_parser& parser, std::size_t num_args, std::size_t args_split) const {
+    void
+    add_arguments(ap::argument_parser& parser, std::size_t num_args, std::size_t args_split) const {
```
Comment (dawidlipiak) [inline test/include/argument_parser_test_fixture.hpp:84]: Unnecessary newline here in 84 line.

File: test/include/argument_parser_test_fixture.hpp [hunk 2]
```diff
@@ -93,7 +94,7 @@ struct argument_parser_test_fixture {
     }
 
     cmd_argument_list
-        prepare_cmd_arg_list(std::size_t num_args, std::size_t args_split) const {
+    prepare_cmd_arg_list(std::size_t num_args, std::size_t args_split) const {
```
Comment (dawidlipiak) [inline test/include/argument_parser_test_fixture.hpp:97]: One more unnecessary newline.

**Review by dawidlipiak — COMMENTED** [review]
All seems fine to me. In file test/include/argument_parser_test_fixture.hpp I personally  would avoid writing function is style like this:
image
If you were trying to make it more wrapped for it to not be such a long one-line code it's better to wrap function arguments like this:
`argument_interface& sut_set_value ( 
     const argument_ptr_type& sut, 
     const std::string& str_value 
) const {
     return sut->value(str_value);
}
`
But again it's just a coding style preference, you don't have to change it.

File: test/include/argument_parser_test_fixture.hpp (hunk 2 above)
Comment (SpectraL519) [reply test/include/argument_parser_test_fixture.hpp:97]: Result of clang-format

File: test/include/argument_test_fixture.hpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+namespace ap_testing {
+
+struct argument_test_fixture {
+    argument_test_fixture() = default;
+    ~argument_test_fixture() = default;
+
+    using argument_ptr_type = std::unique_ptr<argument_interface>;
+
+    argument_interface&
+    sut_set_value(const argument_ptr_type& sut, const std::string& str_value) const {
```
[the remark this answers is not in the document]
Comment (SpectraL519) [reply test/include/argument_test_fixture.hpp:21]: Result of clang-format

**Review by dawidlipiak — APPROVED** [review]

**Review by dawidlipiak — APPROVED** [review]

**Pull request merged**